### ISSUE TITLE
Conclusion about callable contravariance is wrong

### ISSUE THREAD

**Issue by Corwinpro** [rednafi/reflections#288]
This is wrong:

https://github.com/rednafi/reflections/blob/081bc519cd895e28239fa6751f944b084335b53d/content/python/variance_of_generic_types.md?plain=1#L87-L89

Example:

```python
class Employee:
    pass


class Manager:
    def bonus(self) -> int:
        return 42


def calculate_salary(person: Manager) -> int:
    return 100 + person.bonus()


employee = Employee()
manager = Manager()
calculate_salary(manager)  # fine
calculate_salary(employee)  # not fine
```

What the contravariance means is that if I use `calculate_salary` somewhere, and there is another function that uses a supertype of `Manager`:

```python
def calculate_salary_better(person: Employee) -> int:
    return 200
```

then it is type safe to replace `calculate_salary` with `calculate_salary_better`.

**Comment by rednafi** [comment]
@Corwinpro Hi, thank you for opening the issue. I did some read-ups after you raised the issue and I think you're correct. I will update that comment with your explanation and refer to this issue.

**Comment by rednafi** [comment]
But in the example you gave above, `Manager` is not a subtype of `Employee`.

**Comment by Corwinpro** [comment]
Sorry it should be `class Manager(Employee)`

**Comment by rednafi** [comment]
Yep. Makes sense. I think you're right. I'll update the comment and refer to this issue. Thank you for reporting this.
